execute/allocator.go

```diff
@@ -220,3 +219,13 @@ func (a *Allocator) GrowTimes(slice []Time, n int) []Time {
 	a.account(diff, timeSize)
 	return s
 }
+
+// String represents a string stored in some backing byte slice.
+type String struct {
+	offset int
+	len    int
+}
+
+func (s String) Bytes(buf *arrowmem.Buffer) []byte {
+	return buf.Bytes()[s.offset : s.offset+s.len]
+}
```

Comment:
👍 My understanding of this new struct is that it stores the column string size and if a string get too long, we can do something with it